stl/src/filesys.cpp

```diff
@@ -195,7 +195,7 @@ _FS_DLL file_type __CLRCALL_PURE_OR_CDECL _Stat(const wchar_t* _Fname, perms* _P
 
     if (GetFileAttributesExW(_Fname, GetFileExInfoStandard, &_Data)) {
         // get file type and return permissions
-        if (_Pmode != 0) {
+        if (_Pmode != nullptr) {

```

Comment:
Should we use conversion to bool here, aka `if (_Pmode)`?

Would apply throughout

Reply:
Our codebase is currently inconsistent about testing pointers directly (or negated), versus comparing them to `nullptr`. (Unlike integers, where we strongly prefer to explicitly compare them against zero.) I personally like directly testing pointers, because I believe it reads naturally ("if we have pointer, do cool things with it") and avoids the double negation inherent in "not null". However, I think that if we want to consistently test pointers directly, we should do so as a *separate* cleanup change.